phys2bids/physio_obj.py

```diff
@@ -325,6 +325,79 @@ def print_info(self, filename):
 
         LGR.info(info)
 
+    def check_trigger_amount_std(self, chtrig=1, num_timepoints_expected=0, tr=0):
+        from itertools import groupby
+        """
+        finds proper threshold according to twice the std of the trigger channel
+
+        Parameters
+        ----------
+
+        chtrig: int
+            channel with the trigger
+        num_timepoints_expected: int
+            Number of expected triggers (num of TRs in fMRI)
+        tr: float
+            The Repetition Time of the fMRI data.
+
+
+        Notes
+        -----
+        Outcome:
+        self.thr_std:
+            Threshold used by the function to detect trigger points.
+        self.num_timepoints_found: int
+            Property of the `BlueprintInput` class.
+            Contains the number of timepoints found
+            with the automatic estimation.
+        self.timeseries:
+            The property `timeseries` is shifted with the 0 being
+            the time of first trigger.
+        """
+        LGR.info('Counting trigger points according to std method')
+        # Use twice the std of the trigger channel to get the threshold
+        # comparing it to a given threshold.
+        trigger = self.timeseries[chtrig]
+        thr_std = np.mean(trigger) + 2 * np.std(trigger)
+        timepoints = trigger > thr_std
+        num_timepoints_found = len([is_true for is_true, _ in groupby(timepoints,
+                                    lambda x: x != 0) if is_true])
```

Comment:
@rmarkello true, but the `groupby` is necessary because triggers could be several timepoints long, and in that case you want to count it once only. Do you think that
` num_timepoints_found = np.sum(groupby(timepoints, lambda x: x != 0))`
would work?

Reply:
Ahhh, totally missed that _very_ important detail!! No, `np.sum` won't work in that case. The solution @vinferrer came up with works! Ignore this suggestion.